lightning/src/ln/channel.rs

```diff
@@ -3315,11 +3313,11 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			feerate_per_kw: fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Background) as u32,
 			to_self_delay: self.our_to_self_delay,
 			max_accepted_htlcs: OUR_MAX_HTLCS,
-			funding_pubkey: PublicKey::from_secret_key(&self.secp_ctx, self.local_keys.funding_key()),
-			revocation_basepoint: PublicKey::from_secret_key(&self.secp_ctx, self.local_keys.revocation_base_key()),
-			payment_point: PublicKey::from_secret_key(&self.secp_ctx, self.local_keys.payment_key()),
-			delayed_payment_basepoint: PublicKey::from_secret_key(&self.secp_ctx, self.local_keys.delayed_payment_base_key()),
-			htlc_basepoint: PublicKey::from_secret_key(&self.secp_ctx, self.local_keys.htlc_base_key()),
+			funding_pubkey: self.local_keys.pubkeys().funding_pubkey,
```

Comment:
I think I would extract self.local_keys.pubkeys() into a variable above so you don't keep calling this method repeatedly.

Reply:
It should get trivially optimized out since it returns a reference and doesn't copy.

Reply:
Will it not get optimized out if you have an intermediate variable?

Reply:
My point being, if the compiler is gonna produce the same thing anyway, you should extract `self.local_keys.pubkeys()` into a separate variable for better legibility.